pkg/driver/node.go

```diff
@@ -678,16 +683,22 @@ func (d *nodeService) nodePublishVolumeForFileSystem(req *csi.NodePublishVolumeR
 }
 
 // getVolumesLimit returns the limit of volumes that the node supports
-func (d *nodeService) getVolumesLimit() int64 {
+func (d *nodeService) getVolumesLimit(ctx context.Context) int64 {
 	if d.driverOptions.volumeAttachLimit >= 0 {
 		return d.driverOptions.volumeAttachLimit
 	}
-	ebsNitroInstanceTypeRegex := "^[cmr]5.*|t3|z1d"
+
 	instanceType := d.metadata.GetInstanceType()
-	if ok, _ := regexp.MatchString(ebsNitroInstanceTypeRegex, instanceType); ok {
-		return defaultMaxEBSNitroVolumes
+	instanceID := d.metadata.GetInstanceID()
+
+	limit, err := d.cloud.GetVolumeAttachLimit(ctx, instanceID)
+	if err != nil {
+		//CO decides on the number of volume of the same type to be published.
```

Comment:
can you check if the e2e tests are currently doing this? by adding a log line here.

Reply:
I think they are doing this because I don't see the "The volumeAttachLimit for the instance type %q is %d" line being printed in the logs https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_aws-ebs-csi-driver/1075/pull-aws-ebs-csi-driver-external-test-eks/1443331300164374528 and the verbosity is set to 5 https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/hack/values.yaml#L4